tasks/build-extensions/compile-e2e-tests.js

```diff
@@ -0,0 +1,29 @@
+const execSync = require('child_process').execSync;
+const getExtensionDirectoriesSync = require('./get-extension-directories-sync');
+
+const directories = getExtensionDirectoriesSync();
+
+// Only compile end-to-end tests if protractor is installed
+// i.e. if devDependencies are installed
+let protractorInstalled;
+
+try {
+    require('protractor');
+    protractorInstalled = true;
+} catch (e) {
+    protractorInstalled = false;
+}
+
+if (protractorInstalled) {
+    directories.forEach(({extensionRootPath}) => {
+        try {
+            execSync(
+                `cd ${extensionRootPath} && npm run compile-e2e`,
```

Comment:
I think there should be `--if-present` otherwise the error message would be false positive when there are no tests.
and when it fails it should probably return an error at the end so we see that on CI

Reply:
I have added `--if-present` flag, but in general, e2e tests aren't running on extensions. They were running at some point in the past, but the implementation was very minimal and due to some incompatibilities we've disabled it. I'll get to it sometime when refactoring stuff.